flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/FlinkRuleSets.scala

```diff
@@ -49,6 +51,15 @@ object FlinkRuleSets {
   val POST_EXPAND_CLEAN_UP_RULES: RuleSet = RuleSets.ofList(
     EnumerableToLogicalTableScan.INSTANCE)
 
+  /**
+    * Pre-processing for logical optimization.
+    */
+  val PRE_LOGICAL_OPT_RULES: RuleSet = RuleSets.ofList(
+    new FlinkAggregateExtractProjectRule(
```

Comment:
extract this into a static final variable in `FlinkAggregateExtractProjectRule`?